pkg/project/project.go

```diff
@@ -51,6 +51,27 @@ type Project struct {
 // Source returns the source projectfile
 func (p *Project) Source() *projectfile.Project { return p.projectfile }
 
+// SetNamespace updates the project namespace and also updates the underlying project file
+func (p *Project) SetNamespace(namespace *Namespaced) error {
+	err := p.Source().SetNamespace(namespace.String())
+	if err != nil {
+		return errs.Wrap(err, "Failed to write namespace to projectfile")
+	}
+	p.owner = namespace.Owner
+	p.name = namespace.Project
+	return nil
+}
+
+// SetCommit updates the commit ID for the project and also updates the underlying project file
+func (p *Project) SetCommit(commitID string) error {
+	fail := p.Source().SetCommit(commitID)
+	if fail != nil {
+		return errs.Wrap(fail.ToError(), "Failed to write commitID to projectfile")
+	}
+	p.commitID = commitID
+	return nil
+}
```

Comment:
This is only really needed because we are recording these fields to the project struct, which looking at it now it doesn't really need to do at all. Let's instead fix this by sourcing these fields from the sourced projectfile.